scripts/check-license-header.sh

```diff
@@ -0,0 +1,105 @@
+## Licensed under Apache License v2.0
+##
+## See LICENSE.txt for license information
+## See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+##
+## SPDX-License-Identifier: Apache-2.0
+##
+##===----------------------------------------------------------------------===##
+set -euo pipefail
+
+log() { printf -- "** %s\n" "$*" >&2; }
+error() { printf -- "** ERROR: %s\n" "$*" >&2; }
+fatal() { error "$@"; exit 1; }
+
+test -n "${PROJECT_NAME:-}" || fatal "PROJECT_NAME unset"
+
+expected_file_header_template="@@===----------------------------------------------------------------------===@@
+@@
+@@ This source file is part of the ${PROJECT_NAME} open source project
+@@
+@@ Copyright (c) YEARS Apple Inc. and the ${PROJECT_NAME} project authors
+@@ Licensed under Apache License v2.0
+@@
+@@ See LICENSE.txt for license information
+@@ See CONTRIBUTORS.txt for the list of ${PROJECT_NAME} project authors
+@@
+@@ SPDX-License-Identifier: Apache-2.0
+@@
+@@===----------------------------------------------------------------------===@@"
+
+paths_with_missing_license=( )
+
+file_paths=$(git ls-files \
+  ":(exclude).gitignore" \
+  ":(exclude)**/.gitignore" \
+  ":(exclude).licenseignore" \
+  ":(exclude).gitattributes" \
+  ":(exclude).mailfilter" \
+  ":(exclude).mailmap" \
+  ":(exclude).spi.yml" \
+  ":(exclude).swift-format" \
+  ":(exclude).github/*" \
+  ":(exclude)*.md" \
+  ":(exclude)*.txt" \
+  ":(exclude)*.yaml" \
+  ":(exclude)*.json" \
+  ":(exclude)Package.swift" \
+  ":(exclude)**/Package.swift" \
+  ":(exclude)Package.resolved" \
+  ":(exclude)**/Package.resolved" \
+  ":(exclude)Makefile" \
+  ":(exclude)*.modulemap" \
+  ":(exclude)**/*.modulemap" \
+  ":(exclude)**/*.docc/*" \
+  ":(exclude)*.xcprivacy" \
+  ":(exclude)**/*.xcprivacy" \
+  ":(exclude)*.symlink" \
+  ":(exclude)**/*.symlink" \
+  ":(exclude)Dockerfile" \
+  ":(exclude)**/Dockerfile" \
+  ":(exclude)Snippets/*" \
+  $(printf ":(exclude)%s " $(cat .licenseignore))
+)
+
+while IFS= read -r file_path; do
+  file_basename=$(basename -- "${file_path}")
+  file_extension="${file_basename##*.}"
+
+  case "${file_extension}" in
+    swift) expected_file_header=$(sed -e 's|@@|//|g' <<<"${expected_file_header_template}") ;;
+    h) expected_file_header=$(sed -e 's|@@|//|g' <<<"${expected_file_header_template}") ;;
+    c) expected_file_header=$(sed -e 's|@@|//|g' <<<"${expected_file_header_template}") ;;
+    yml) expected_file_header=$(sed -e 's|@@|##|g' <<<"${expected_file_header_template}") ;;
+    sh) expected_file_header=$(cat <(echo '#!/bin/bash') <(sed -e 's|@@|##|g' <<<"${expected_file_header_template}")) ;;
+    py) expected_file_header=$(cat <(echo '#!/usr/bin/env python3') <(sed -e 's|@@|##|g' <<<"${expected_file_header_template}")) ;;
+    rb) expected_file_header=$(cat <(echo '#!/usr/bin/env ruby') <(sed -e 's|@@|##|g' <<<"${expected_file_header_template}")) ;;
+    *) fatal "Unsupported file extension for file (exclude or update this script): ${file_path}" ;;
```

Comment:
Should we ignore all json since it can't have a header as is likely used for config/input for a bunch of things?

Reply:
JSON is ignored via the standard excludes so we don't have to handle it.

Reply:
Ah, missed that one!

Reply:
Let’s not have standard excludes. Too easy for things to accidentally creep through.